lib/addons/opa-gatekeeper/index.ts

```diff
@@ -0,0 +1,87 @@
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+
+/**
+ * Properties available to configure opa gatekeeper
+ */
+export interface OpaGatekeeperAddOnProps {
+    /**
+     * Create namespace for Gatekeeper controller 
+     * @default true
+     */
+    createNamespace?: boolean;
+
+    /** Default namespace
+     * @default gatekeeper-system
+     */
+    namespace?: string;
+
+    /**
+     * Helm chart version to use to install.
+     * @default 3.7.0-beta.1
+     */
+    version?: string;
+
+    /**
+     * Setting validatingWebhookFailurePolicy to True
+     * @default true 
+     */
+    disableValidatingWebhook?: boolean;
+
+    /**
+     * Label namespace so that webhook doesn't interfere with deployment of gatekeeper chart
+     * @default true
+     */
+    labelNamespace?: boolean;
+
+    /**
+     * values to pass to the helm chart per https://github.com/open-policy-agent/gatekeeper/tree/master/charts/gatekeeper
+     */
+     values?: {
+        [key: string]: any;
+    };
+}
+
+/**
+ * Defaults options for the gatekeeper add-on
+ */
+
+const defaultProps: OpaGatekeeperAddOnProps = {
+    namespace: 'gatekeeper-system',
+    version: '3.7.0-beta.1',
+    disableValidatingWebhook: false,
+    labelNamespace: true,
+    values: {}
+};
+
+
+export class OpaGatekeeperAddOn implements ClusterAddOn {
+    
+    readonly options: OpaGatekeeperAddOnProps;
+    
+    constructor(props?: OpaGatekeeperAddOnProps) {
+        this.options = { ...defaultProps, ...props };
+    }
+
+    deploy(clusterInfo: ClusterInfo): void {
+
+        const props = this.options;
+
+        const values = { ...props.values ?? {}};
+
+        values['gatekeeper-configurations'] = {
```

Comment:
How did you come up with 'gatekeeper-configurations' and 'configurations' attributes? This will produce values like this:
```yaml
gatekeeper-configurations:
  configurations:
     disableValidatingWebhook: false
     labelNamespace: true
```

You need:
```yaml
disableValidatingWebhook: false
labelNamespace: true
```

Reply:
removed and took a different approach and as shown in my latest push